vim/core/oni-plugin-buffers/index.js

```diff
@@ -0,0 +1,109 @@
+// @ts-check
+const path = require('path');
+
+const activate = Oni => {
+  const menu = Oni.menu.create();
+
+  const truncateFilePath = filepath => {
+    const sections = filepath.split(path.sep);
+    const folderAndFiles = sections.slice(-2);
+    return folderAndFiles.join(path.sep);
+  };
+
+  const updateBufferList = (Oni, menu) => {
+    const buffers = Oni.editors.activeEditor.getBuffers();
+    const active = Oni.editors.activeEditor.activeBuffer.filePath;
+
+    const bufferMenuItems = buffers.map(b => ({
+      label: `${active === b.filePath ? b.id + ' %' : b.id}`,
+      detail: truncateFilePath(b.filePath),
+      icon: Oni.ui.getIconClassForFile(b.filePath),
+      pinned: active === b.filePath,
+      metadata: { filePath: b.filePath, id: b.id },
+    }));
+
+    return bufferMenuItems;
+  };
+
+  const createBufferList = () => {
+    const buffers = updateBufferList(Oni, menu);
+    menu.show();
+    menu.setItems(buffers);
+  };
+
+  const toggleBufferList = () => {
+    !menu.isOpen() ? createBufferList() : menu.hide();
+  };
+
+  const deleteBuffer = menu => {
+    if (menu.selectedItem) {
+      Oni.editors.activeEditor.bufferDelete(menu.selectedItem.metadata.id);
+      menu.hide();
+    }
+  };
+
+  const openBuffer = (menu, orientation) => {
+    if (menu.selectedItem && menu.isOpen()) {
+      const buffers = Oni.editors.activeEditor.getBuffers();
+      try {
+        const { filePath = '' } = menu.selectedItem.metadata;
+        Oni.editors.activeEditor.openFile(filePath, orientation);
+      } catch(e) {
+        console.warn('[Oni Buffer Plugin Error]: ', e);
+      } finally {
+        menu.hide();
+      }
+    }
+    return;
+  };
+
+  Oni.commands.registerCommand({
+    command: 'bufferlist.delete',
+    name: 'Delete Selected Buffer',
+    execute: () => deleteBuffer(menu),
+  });
+
+  Oni.commands.registerCommand({
+    command: 'bufferlist.split',
+    name: 'Split Selected Buffer',
+    execute: () => openBuffer(menu, 'horizontal'),
+  });
+
+  Oni.commands.registerCommand({
+    command: 'bufferlist.vsplit',
+    name: 'Vertical Split Selected Buffer',
+    execute: () => openBuffer(menu, 'vertical'),
```

Comment:
This is where I was mentioning you can add an `enabled` filter - like - so you could bake in the filter to the command, instead of having the user needing to know to apply the filter. It definitely is more convenient to do it here, so that we always do the 'right' thing - and the user doesn't need to think hard about it.